### ISSUE TITLE
Kafka Consumer is disconnected after some hours of inactivity (no reconnect).

### ISSUE THREAD

**Issue by spyros87** [yandex/ClickHouse#5680]
**Describe the bug**
After some time of inactivity CH Kafka consumer gets disconnected from Broker. However, it is not able to connect again and start consume from the assigned group.

**How to reproduce**
* **Version** 19.7.3.9
* 2-Partition Kafka topic, 2 CH cluster with replicated persistence table. Each CH consumes from 1 partition.
* Query for constructing Kafka Consumer
```
CREATE TABLE test.queue (
          `record_date`      DateTime,
          `record_timestamp` UInt64,
          `name`             String,
          `alias`            String,
          `domain`           String
) ENGINE = Kafka 
  SETTINGS kafka_broker_list = 'broker-1:9092,broker-2:9092',
           kafka_topic_list = 'test-topic',
           kafka_group_name = 'test-group',
           kafka_format = 'JSONEachRow',
           kafka_num_consumers = 1,
           kafka_row_delimiter = '\n',
           kafka_skip_broken_messages = 0;
```

**Expected behavior**
After connection to Kafka is dropped, the consumer can retry and reconnect.

**Error message and/or stacktrace**
CH Log Trace:
```
2019.06.18 16:59:53.869237 [ 24 ] {} <Error> StorageKafka (queue): Consumer error: Local: Broker transport failure
2019.06.18 16:59:53.869290 [ 24 ] {} <Trace> StorageKafka (queue): Committing message with offset 0
2019.06.18 16:59:53.869483 [ 24 ] {} <Trace> StorageKafka (queue): Re-joining claimed consumer after failure
2019.06.18 16:59:53.869835 [ 24 ] {} <Error> void DB::StorageKafka::streamThread(): std::exception. Code: 1001, type: cppkafka::HandleException, e.what() = Local: Invalid argument or configuration, version = 19.7.3.9 (official build)
**(Last message repeats itself)**
```

Kafka Log Trace:
```
[2019-06-18 16:59:55,356] INFO [GroupCoordinator 56]: Preparing to rebalance group tick-group in state PreparingRebalance with old generation 9435 (__consumer_offsets-31) (reason: Updating metadata for member ClickHouse 19.7.3.9-beb10d98-bc06-458e-bc98-eb39f2999c20) (kafka.coordinator.group.GroupCoordinator)
[2019-06-18 16:59:56,468] INFO [GroupCoordinator 56]: Stabilized group tick-group generation 9436 (__consumer_offsets-31) (kafka.coordinator.group.GroupCoordinator)
[2019-06-18 16:59:56,475] INFO [GroupCoordinator 56]: Assignment received from leader for group tick-group for generation 9436 (kafka.coordinator.group.GroupCoordinator)
[2019-06-18 16:59:59,873] INFO [GroupCoordinator 56]: Member ClickHouse 19.7.3.9-beb10d98-bc06-458e-bc98-eb39f2999c20 in group tick-group has left, removing it from the group (kafka.coordinator.group.GroupCoordinator)
[2019-06-18 16:59:59,873] INFO [GroupCoordinator 56]: Preparing to rebalance group tick-group in state PreparingRebalance with old generation 9436 (__consumer_offsets-31) (reason: removing member ClickHouse 19.7.3.9-beb10d98-bc06-458e-bc98-eb39f2999c20 on LeaveGroup) (kafka.coordinator.group.GroupCoordinator)
[2019-06-18 17:00:02,468] INFO [GroupCoordinator 56]: Stabilized group tick-group generation 9437 (__consumer_offsets-31) (kafka.coordinator.group.GroupCoordinator)
[2019-06-18 17:00:02,470] INFO [GroupCoordinator 56]: Assignment received from leader for group tick-group for generation 9437 (kafka.coordinator.group.GroupCoordinator)
[2019-06-18 17:00:02,490] INFO [GroupCoordinator 56]: Preparing to rebalance group tick-group in state PreparingRebalance with old generation 9437 (__consumer_offsets-31) (reason: Adding new member ClickHouse 19.7.3.9-dd041051-dcb4-4df6-a9df-f46459ab16f1) (kafka.coordinator.group.GroupCoordinator)
[2019-06-18 17:00:05,468] INFO [GroupCoordinator 56]: Stabilized group tick-group generation 9438 (__consumer_offsets-31) (kafka.coordinator.group.GroupCoordinator)
[2019-06-18 17:00:05,469] INFO [GroupCoordinator 56]: Preparing to rebalance group tick-group in state PreparingRebalance with old generation 9438 (__consumer_offsets-31) (reason: Adding new member ClickHouse 19.7.3.9-673f57ba-7d9c-442d-8106-be9ac2c67c5e) (kafka.coordinator.group.GroupCoordinator)
[2019-06-18 17:00:15,468] INFO [GroupCoordinator 56]: Member ClickHouse 19.7.3.9-dd041051-dcb4-4df6-a9df-f46459ab16f1 in group tick-group has failed, removing it from the group (kafka.coordinator.group.GroupCoordinator)
**(Last message repeats itself as the message above)**
```

**Comment by spyros87** [comment]
This issue can be fixed if librdkafka can be upgraded to v1.0.1 according to https://github.com/edenhill/librdkafka/issues/2371

**Comment by filimonov** [comment]
https://github.com/yandex/ClickHouse/pull/5699

**Comment by spyros87** [comment]
@filimonov I have already seen that this PR has been merged. I am waiting for a stable 19.11 in order to validate. Thank you for sharing this info.

**Comment by filimonov** [comment]
@spyros87 can you check 19.11.7?

**Comment by abyss7** [comment]
Should be fixed in 19.14+. Possibly was backported to 19.11 and 19.13
